src/models/oe-zoneerder-config.ts

```diff
@@ -0,0 +1,137 @@
+import { LayerType } from '@/models';
+import type { ILocatie, UrlString } from '@/models';
+import type { Contour } from '@models/oe-openlayers';
+
+export interface ControlConfig {
+  fullscreen: boolean;
+  zoomInOut: boolean;
+  zoomFullExtent: boolean;
+  zoomGeoLocation: boolean;
+  rotate: boolean;
+  zoomSwitcher: boolean;
+}
+
+export const defaultControlConfig: ControlConfig = {
+  fullscreen: true,
+  zoomInOut: true,
+  zoomFullExtent: false,
+  zoomGeoLocation: false,
+  rotate: false,
+  zoomSwitcher: false,
+};
+
+interface BaseLayerOptions {
+  type: LayerType;
+  title: string;
+  visible?: boolean;
+  legendImages?: legendImage[];
+}
+
+interface legendImage {
+  title: string;
+  url: string;
+}
+
+interface GrbOrNgiLayerOptions extends BaseLayerOptions {
+  type: LayerType.Ngi | LayerType.GRB;
+}
+
+interface WmsLayerOptions extends BaseLayerOptions {
+  wmsLayers: string;
+}
+
+export interface GrbWmsLayerOptions extends WmsLayerOptions {
+  type: LayerType.GrbWMS;
+}
+
+export interface ErfgoedWmsLayerOptions extends WmsLayerOptions {
+  type: LayerType.ErfgoedWms;
+}
+
+export interface DHMVLayerOptions extends BaseLayerOptions {
+  type: LayerType.DHMV;
+}
+
+export interface OMWRGBMRVLOptions extends BaseLayerOptions {
+  type: LayerType.OMWRGBMRVL;
+}
+
+export type GrbLayerOptions = GrbOrNgiLayerOptions;
+export type NgiLayerOptions = GrbOrNgiLayerOptions;
+
+export type LayerOptions =
+  | GrbLayerOptions
+  | NgiLayerOptions
+  | GrbWmsLayerOptions
+  | ErfgoedWmsLayerOptions
+  | DHMVLayerOptions
+  | OMWRGBMRVLOptions;
+
+export interface LayerConfig {
+  baseLayers: { [layerId: string]: LayerOptions };
+  overlays: { [layerId: string]: LayerOptions };
+}
+
+const legendParams = `?service=WMS&request=GetLegendGraphic&format=image%2Fpng&width=20&height=20&layer=`;
+const legendBaseUrl = `https://geo.onroerenderfgoed.be/geoserver/wms${legendParams}`;
```

Comment:
liefst tegen test-geoserver werken

Reply:
Hmm dit is de default config, dat zou imho wel productie moeten zijn. maar mss dit in een variable steken zodat dit gemakkelijker overridden kan worden zonder de volledige layerconfig te moeten overschrijven

Reply:
ahja check dacht dat dit een story was. idd liefst in aparte config zoals beschermingenWmsUrl